app/modules/auth/Auth.tsx

```diff
@@ -5,18 +5,29 @@ import names from 'classnames'
 import cl from './Auth.module.scss'
 import { AuthInputs } from './types/auth.types'
 import { User } from 'firebase/auth'
+import { useRouter } from 'next/navigation'
+import { LOCAL_STORAGE_USER_ID_KEY } from './constants'
 
 type Props = {
   onSubmit: ({ email, password }: AuthInputs) => Promise<undefined | User>
 }
 
 const Auth = ({ onSubmit }: Props) => {
```

Comment:
I'm not entrelly sure why you pass `onSubmit` here, it's a bit weird, but maybe there is a reason for this, let's disscuss.

Reply:
onSubmit can be AuthApi.login or AuthApi.register depending on the page